pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -124,3 +127,73 @@ func replaceParamValues(params []v1beta1.Param, stringReplacements map[string]st
 	}
 	return params
 }
+
+// ApplyTaskResultsToPipelineResults applies the results of completed TasksRuns to a Pipeline's
+// list of PipelineResults, returning the computed set of PipelineRunResults. References to
+// non-existent TaskResults or failed TaskRuns result in a PipelineResult being considered invalid
+// and omitted from the returned slice. A nil slice is returned if no results are passed in or all
+// results are invalid.
```

Comment:
i apologize for probably having already asked this, but it seems weird to me that we just ignore the invalid values, feels like we should be returning an error?

Reply:
I'm not sure it's quite right to say that we ignore the invalid values.  We return a `nil` pointer if the value is invalid - in other words, no valid value was found for the result reference.

I don't totally agree that this is an error state but I don't feel strongly enough to dispute it.  What should we do with that error when it's returned?  In the old code it was silently ignored.

Reply:
> feels like we should be returning an error?

Returning an error might break `pipelineRuns`. There is no error reported in logs or to users at all. If we introduce error now, `pipelineRuns` which used to be successful will start failing with this error. And at the same time, the least we could do is log the error if needed.

Reply:
We should eventually move the execution of this logic to before the pipelinerun state is set.
That way could still fail the pipeline if results are broken or invalid in some way.
Maybe something for next release.